src/librustc_resolve/lib.rs

```diff
@@ -2194,7 +2177,9 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
     // user and one 'x' came from the macro.
     fn binding_mode_map(&mut self, pat: &Pat) -> BindingMap {
         let mut result = HashMap::new();
-        pat_bindings(&self.def_map, pat, |binding_mode, _id, sp, path1| {
+        pat_bindings(&self.def_map,
+                     pat,
+                     |binding_mode, _id, sp, path1| {
```

Comment:
I don't know what rules this follows, but in this specific example (and a few ones below), I think that the closure "header" should rather be on the same line as the other parameters.